src/controllers/currentAccount.ts

```diff
@@ -325,4 +325,47 @@ export class CurrentAccountController {
       [`balance_${address}`]: balance,
     });
   }
+
+  async updateOtherAccountsBalances() {
+    const currentNetwork = this.getNetwork();
+    const url = new URL('addresses/balance', this.getNode());
+    const addresses = this.getAccounts().map(account => account.address);
+
+    while (addresses.length > 0) {
+      const splicedAddresses = addresses.splice(0, 1000);
+      const response = await fetch(url, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          addresses: splicedAddresses,
+        }),
+      });
+
+      const regularBalances = (await response.json()) as Array<{
+        id: string;
+        balance: string;
+      }>;
+
+      const balances = regularBalances.reduce<Record<string, BalancesItem>>(
+        (acc, regularBalance) => {
+          const balanceKey = `balance_${regularBalance.id}`;
+          const existingBalance = this.store.getState()[balanceKey];
```

Comment:
May be separate `this.store.getState()` and getting a key in it? It would be more readable. And this way it's trivial to call `this.store.getState()` just once outside of the loop, instead of calling it on every iteration.